src/zope/testing/doctestcase.txt

```diff
@@ -0,0 +1,219 @@
+methods.
+
+You can define doctests in 4 ways:
+
+- references to named files
+
+- strings
+
+- decorated functions with docstrings
+
+- reference to named files decorating test-specific setup functions
+
+.. some setup
+
+   >>> __name__ = 'tests'
+
+Here are some examples::
+
+    >>> from zope.testing import doctestcase
+    >>> import doctest
+    >>> import unittest
+
+    >>> g = 'global'
+    >>> class MyTest(unittest.TestCase):
+    ...
+    ...     def setUp(self):
+    ...         self.a = 1
+    ...         self.globs = dict(c=9)
+    ...
+    ...     test1 = doctestcase.file('test1.txt', optionflags=doctest.ELLIPSIS)
+    ...
+    ...     test2 = doctestcase.docteststring('''
+    ...       >>> self.a, g, c
+    ...       (1, 'global', 9)
+    ...     ''')
+    ...
+    ...     @doctestcase.doctestmethod(optionflags=doctest.ELLIPSIS)
+    ...     def test3(self):
+    ...         '''
+    ...         >>> self.a, self.x, g, c
+    ...         (1, 3, 'global', 9)
+    ...         '''
+    ...         self.x = 3
+    ...
+    ...     @doctestcase.doctestfile('test4.txt')
+    ...     def test4(self):
+    ...         self.x = 5
+
+.. We can run these tests with the ``unittest`` test runner.
+
+    >>> loader = unittest.TestLoader()
+    >>> suite = loader.loadTestsFromTestCase(MyTest)
+    >>> import sys
+    >>> sys.stdout.writeln = lambda s: sys.stdout.write(s+'\n')
+    >>> result = suite.run(unittest.TextTestResult(sys.stdout, True, 3))
+    test1 (tests.MyTest) ... ok
+    test2 (tests.MyTest) ... ok
+    test3 (tests.MyTest) ... ok
+    test4 (tests.MyTest) ... ok
+
+
+    >>> for _, e in result.errors:
+    ...     print(e); print
+
+In this example, 3 constructors were used:
+
+doctestfile (alias: file)
+  doctestfile makes a file-based test case.
+
+  This can be used as a decorator, in which case, the decorated
+  function is called before the test is run, to provide test-specific
+  setup.
+
+docteststring (alias string)
+  docteststring constructs a doctest from a string.
+
+doctestmethod (alias method)
+  doctestmethod constructs a doctest from a method.
+
+  The method's docstring provides the test. The method's body provides
+  optional test-specific setup.
+
+Note that short aliases are provided, which maye be useful in certain
+import styles.
+
+Tests have access to the following data:
+
+- Tests created with the ``docteststring`` and ``doctestmethod``
+  constructors have access to the module globals of the defining
+  module.
+
+- In tests created with the ``docteststring`` and ``doctestmethod``
+  constructors, the test case instance is available as the ``self``
+  variable.
+
+- In tests created with the ``doctestfile`` constructor, the test case
+  instance is available as the ``test`` variable.
+
+- If a test case defines a globs attribute, it must be a dictionary
+  and it's contents are added to the test globals.
+
+The constructors accept standard doctest ``optionflags`` and
+``checker`` arguments.
```

Comment:
Might be worth mentioning somewhere that IGNORE_EXCEPTION_DETAIL is always implicitly OR'ed in.